react/react_forms/README.md

```diff
@@ -1,387 +1,477 @@
+
+We can, actually, use the **same function** to handle both of these inputs, like so:
+
+```js
+handleTextChange = (e) => {
+  const { title, value } = e.target;
+  this.setState({
+    [title]: value,
+  })
+}
 ```
 
-Do we really have to write a whole hook for every form field we want to update? That seems repetitive.
+You'll notice that we added a `title` attribute to our text `input` tags in our JSX. These titles correspond directly to the parts of state that we'd like to update with our input's `value`. Conveniently, both of these attributes exist on our event's `target` key. 
+
+Using a special type of square-bracket syntax (`[title]: value`), we can set a variable as a key in an object. When a user, Damien, types their first name into the corresponding field, we pass the `title` of the field in-between the square brackets, and so doing comes up with something like this when `onChange` is called and the values are passed in:
+
+```js
+this.setState({
+  firstName: "Damien"
+})
+```
 
-Luckily there are other options! We can write a [custom hook](https://serverless-stack.com/chapters/create-a-custom-react-hook-to-handle-form-fields.html). We can also store the state values in one hook and make our function [use dynamic key names](https://stackoverflow.com/questions/55757761/handle-an-input-with-react-hooks) (see #2).
+This way, even though we don't reference the key directly, we make sure that the "First Name" field adjusts the `firstName` part of state, and so on.
 
-## 4. Handling submission
+### Handling submission
 
-Since we've done everything in a `controlled` manner, we have all the values that we need in our component's state. So all we have to do is modify our `onSubmit` function to grab the variables.
+Since we've done everything in a controlled manner, we have all the values that we need in our component's state. All we have to do now is modify our `onSubmit` function to get these values from our state and do what we want with them.
 
-When the submit button is pressed, `handleFormSubmit` is called, and reads from the state to ensure that the form can be submitted. In a future lesson, we'll see how to react to the state changing to ensure that the user isn't able submit the form until all fields are valid.
+When the `submit` button is pressed, `handleFormSubmit` is called, and reads from the state to ensure that the form can be submitted. In a future lesson, we'll see how to react to the state changing to ensure that the user isn't able submit the form until all fields are valid.
 
 ```jsx
-const allFieldsValid = () => {
-  return (
-    notARobot &&
-    title &&
-    firstName &&
-    lastName
-  );
-};
+allFieldsValid = () => {
+  const { notARobot, userTitle, firstName, lastName } = this.state;
+  return notARobot && userTitle && firstName && lastName;
+}
```

Comment:
In a future lesson, we'll see how to react to the state changing to ensure that the user isn't able submit the form until all fields are valid.

```jsx
allFieldsValid = () => {
  const { notARobot, userTitle, firstName, lastName } = this.state;
  return notARobot && userTitle && firstName && lastName;
}

Isn't that what this is doing?

Reply:
Agree, let's cut the sentence `In a future lesson .... until all fields are valid.` Better in general not to promise content for future lessons because that will make it easier for us to reorder content in the future.